src/main/java/com/google/sps/servlets/Util.java

```diff
@@ -173,13 +173,26 @@ public void associateRefreshToken(String userId, String refreshToken) {
 
   public void updateDevices(String userId, List<String> deviceIds, String updatesInJson) throws IOException {
     final String accessToken = getAccessToken(userId);
-    for (final String deviceId : deviceIds) {
-      final String updateURL = getUpdateUrl(deviceId);
-      RequestBody body = RequestBody.create(JSON_TYPE, updatesInJson);
-      Request req = new Request.Builder().url(updateURL).put(body).addHeader("Authorization", "Bearer " + accessToken).build();
-      Response updateResponse = client.newCall(req).execute();
-      updateResponse.body().close();
-    }
+    deviceIds
+      .parallelStream()
+      .forEach(
+        deviceId -> {
+          try {
+            updateSingleDevice(accessToken, deviceId, updatesInJson);
+          } catch (IOException e) {
+            System.out.println(deviceId);
+            System.out.println("had an error");//TODO: handle and return failed devices
+          }
```

Comment:
I do not know much about the API, but if we get a response back is it safe to assume the device has updated successfully as is done here?  Or are there cases where we get a response with an error (if we somehow try to update an invalid device ID or don't have correct permissions) and if so how can we handle them.

Reply:
It is not safe to assume that. Will update this to respond with a list of device IDs/serial numbers of devices that have failed.

Reply:
for now it only returns a list of device Ids